### PR TITLE
fix(MsgPublish): support remove duplicates for payload & headers

### PR DESCRIPTION
**Pull request by oceanlvr**
### PR Checklist

If you have any questions, you can refer to the [Contributing Guide](https://github.com/emqx/MQTTX/blob/master/.github/CONTRIBUTING.md)

#### Issue Number

None

#### What is the new behavior?

support remove duplicates for payload & headers.

#### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

#### Specific Instructions

Are there any specific instructions or things that should be known prior to reviewing?

### CODE REVIEW

File: src/utils/mqttUtils.ts [hunk 1]
```diff
@@ -131,4 +133,25 @@ export const getMQTTProtocol = (data: ConnectionModel): Protocol => {
...
+}
+
+export const hasMessageHeader = async (data: HistoryMessageHeaderModel): Promise<boolean> => {
+  const headers = await loadHistoryMessageHeaders()
+  const res = headers.some((el: HistoryMessageHeaderModel) => {
+    return (
+      data.connectionId === el.connectionId &&
+      data.qos === el.qos &&
+      data.topic === el.topic &&
+      data.retain === el.retain
```
Comment (ysfscream) [inline src/utils/mqttUtils.ts:151]: There are too many logical operations here, how about concatenating them as strings?

File: src/components/MsgPublish.vue [hunk 2]
```diff
@@ -191,8 +198,8 @@ export default class MsgPublish extends Vue {
       topic: this.msgRecord.topic,
       qos: this.msgRecord.qos,
     } as HistoryMessageHeaderModel
-    await createHistoryMessagePayload(payload)
-    await createHistoryMessageHeader(header)
+    !(await hasMessagePayload(payload)) && (await createHistoryMessagePayload(payload))
+    !(await hasMessageHeader(header)) && (await createHistoryMessageHeader(header))
```
Comment (ysfscream) [inline src/components/MsgPublish.vue:202]: I am not very clear what is done here.

Comment (oceanlvr) [reply src/components/MsgPublish.vue:202]: I want to insert a payload message if it hasn't been inserted.

```javascript
const res = await hasMessagePayload(payload);
if (res) {
  await createHistoryMessagePayload(payload)
}
```

File: src/utils/mqttUtils.ts (hunk 1 above)
Comment (oceanlvr) [reply src/utils/mqttUtils.ts:151]: I think there should not concatenating as a single string because the `===` operator makes sure the type currently.

**Review by ysfscream — APPROVED** [review]

**Pull request merged**